File: 4-curso/2 cuatrimestre/DS/Practica 1/Ejercicios/Ejercicios 2/ejercicio2s2/src/main/java/com/p1/GestorFiltros.java
package com.p1;

public class GestorFiltros{
    private ServerTarget target;
    private CadenaFiltros cadena;

    GestorFiltros() {
        cadena = new CadenaFiltros();
    }

    GestorFiltros(CadenaFiltros cad) {
        cadena = cad;
    }

    GestorFiltros(ServerTarget servidor) {
        target = servidor;
        cadena = new CadenaFiltros();
    }

    public String aniadeComentario(String comentario) {
        // Añade la cadena original al "diccionario" de cadenas
        target.aniadeComentario(comentario);
        
        // Procesamiento de la cadena
        return cadena.ejecutar(comentario);
    }

    public void aniadirFiltro(Filtro filtro) {
        cadena.aniadirFiltro(filtro);
    }



}

File: 4-curso/2 cuatrimestre/DS/Practica 2/ejercicio_filtro/lib/GestorFiltros.dart
import 'package:ejercicio_filtro/Filtro.dart';
import 'package:tuple/tuple.dart';

import 'ServerTarget.dart';
import 'CadenaFiltros.dart';
import 'FiltroHora.dart';

class GestorFiltros {
  late ServerTarget _target;
  late CadenaFiltros _cadena;

  GestorFiltros() {
    _cadena = CadenaFiltros();
    _target = ServerTarget();

    aniadirFiltro(FiltroHora());
  }

  GestorFiltros.n2(CadenaFiltros cad) {
    _cadena = cad;
    _target = ServerTarget();
  }

  GestorFiltros.n3(ServerTarget servidor) {
    _target = servidor;
    _cadena = CadenaFiltros();

    aniadirFiltro(FiltroHora());
  }

  void aniadeComentario(String comentario) {
    
    _target.aniaddeComentario(comentario, _cadena.ejecutar(comentario));
  }

  Tuple2 getComentario(int i) {
    return _target.getComentarios()[i];
  }

  List<Tuple2> getComentarios() {
    return _target.getComentarios();
  }

  CadenaFiltros getCadena() {
    return _cadena;
  }

  void aniadirFiltro(Filtro filtro) {
    _cadena.aniadeFiltro(filtro);
  }

  void resetGestor() {
    _cadena.resetCadena();

    aniadirFiltro(FiltroHora());
  }
}
